### PR TITLE
Tools: Topology2: Add IIR, FIR, DRC to HDA analog playback

### PR DESCRIPTION
**Pull request by singalsu**
This patch replaces pipeline mixout-gain-dai-copier-playback with mixout-gain-efx-dai-copier-playback. The purpose is to allow enhance of audio playback quality with equalizers (IIR and FIR) and dynamic range control (DRC).

The components are programmed by default to passthrough for minimal overhead. They can be activated via sof-ctl or UCM with ALSA controls.

- Post Mixer Analog Playback IIR Eq bytes
- Post Mixer Analog Playback FIR Eq bytes
- Post Mixer Analog Playback DRC bytes
- Post Mixer Analog Playback DRC switch

The recommendation is to keep headphone playback flat and enable processing only when HDA output is in speaker mode.

### CODE REVIEW

File: tools/topology/topology2/cavs-mixin-mixout-hda.conf [hunk 1]
```diff
@@ -43,6 +46,26 @@ Object.Pipeline {
 					name 'Post Mixer $ANALOG_PLAYBACK_PCM Volume'
 				}
 			}
+			Object.Widget.eqiir.1 {
+				Object.Control.bytes."1" {
+					name 'Post Mixer $ANALOG_PLAYBACK_PCM IIR Eq'
```
Comment (singalsu) [inline tools/topology/topology2/cavs-mixin-mixout-hda.conf:51]: Should append bytes to controls names for IIR and FIR, like there is for DRC.

**Comment by singalsu** [comment]
Topology before and after pictures:
before-sof-hda-generic-4ch
after-sof-hda-generic-4ch

**Comment by singalsu** [comment]
> Wouldn't changing HDA_CONFIG=mix to HDA_CONFIG=efx in generic hda topology build rules do mostly the same thing as this PR? Thou there is a bit more stuff in cavs-mixin-mixout-efx-hda.conf than in this PR.

Yes you are right. I was thinking to keep efx as development topology, with more features added soon, and copy them eventually to default topology names once the new features are considered stabilized.

**Review by jsarha — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
Known pause-resume seen on https://sof-ci.01.org/sofpr/PR8934/build3325/devicetest/index.html . In https://sof-ci.01.org/sofpr/PR8934/build3326/devicetest/index.html otherwise good, but multicore tests hit https://github.com/thesofproject/sof/issues/8942

**Pull request merged**